build/seqan-config.cmake

```diff
@@ -0,0 +1,433 @@
+        set (SEQAN_LIBRARIES ${SEQAN_LIBRARIES} execinfo elf)
+    endif ()
+endif (_SEQAN_HAVE_EXECINFO)
+
+# ----------------------------------------------------------------------------
+# ZLIB dependency
+# ----------------------------------------------------------------------------
+
+if (NOT SEQAN3_NO_ZLIB)
+    find_package (ZLIB ${SeqAn_FIND_QUIETLY})
+endif ()
+
+if (ZLIB_FOUND)
+    set (SEQAN_LIBRARIES         ${SEQAN_LIBRARIES}         ${ZLIB_LIBRARIES})
+    set (SEQAN_INCLUDE_DIRS      ${SEQAN_INCLUDE_DIRS}      ${ZLIB_INCLUDE_DIRS})
+    set (SEQAN_DEFINITIONS       ${SEQAN_DEFINITIONS}       "-DSEQAN3_HAS_ZLIB=1")
+endif ()
+
+# ----------------------------------------------------------------------------
+# BZip2 dependency
+# ----------------------------------------------------------------------------
+
+if (NOT SEQAN3_NO_BZIP2)
+    find_package (BZip2 ${SeqAn_FIND_QUIETLY})
+endif ()
+
+if (NOT ZLIB_FOUND AND BZIP2_FOUND)
+    # NOTE(marehr): iostream_bzip2 uses the type `uInt`, which is defined by
+    # `zlib`. Therefore, `bzip2` will cause a ton of errors without `zlib`.
+    message (AUTHOR_WARNING "Disabling BZip2 [which was successfully found], "
+                            "because ZLIB was not found. BZip2 depends on ZLIB.")
+    unset (BZIP2_FOUND)
+endif ()
+
+if (BZIP2_FOUND)
+    set (SEQAN_LIBRARIES         ${SEQAN_LIBRARIES}         ${BZIP2_LIBRARIES})
+    set (SEQAN_INCLUDE_DIRS      ${SEQAN_INCLUDE_DIRS}      ${BZIP2_INCLUDE_DIRS})
+    set (SEQAN_DEFINITIONS       ${SEQAN_DEFINITIONS}       "-DSEQAN3_HAS_BZIP2=1")
+endif ()
+
+# ----------------------------------------------------------------------------
+# Final check of compilability of seqan3/core/platform.hpp
+# ----------------------------------------------------------------------------
+
+if (NOT DEFINED ${SeqAn_FIND_QUIETLY})
+    message (STATUS "Performing build and dependency checks in <seqan3/core/platform.hpp>")
+endif ()
+
+set(CXXSTD_TEST_SOURCE
+    "#include <seqan3/core/platform.hpp>
+    int main() {}")
+
+# using try_compile instead of check_cxx_source_compiles to capture output in case of failure
+file(WRITE
+     "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/src.cxx"
+     "${CXXSTD_TEST_SOURCE}\n")
+try_compile(SEQAN_PLATFORM_TEST
+            ${CMAKE_BINARY_DIR}
+            ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/src.cxx
+            CMAKE_FLAGS         "-DCOMPILE_DEFINITIONS:STRING=${CMAKE_CXX_FLAGS} ${SEQAN_CXX_FLAGS}"
+                                "-DINCLUDE_DIRECTORIES:STRING=${CMAKE_INCLUDE_PATH} ${SEQAN_INCLUDE_DIRS}"
+            COMPILE_DEFINITIONS ${SEQAN_DEFINITIONS}
+            LINK_LIBRARIES      ${SEQAN_LIBRARIES}
+            OUTPUT_VARIABLE     SEQAN_PLATFORM_TEST_OUTPUT)
+
+if (NOT SEQAN_PLATFORM_TEST)
+    message (STATUS "Performing build and dependency checks in <seqan3/core/platform.hpp> – Failed:\n${SEQAN_PLATFORM_TEST_OUTPUT}")
```

Comment:
I haven't figured out the exact implications, yet. Strictly speaking everything should be STATUS, I think, because we shouldn't interrupt the build. Instead the build gets interrupted when we return without setting SEQAN3_FOUND. **iff** it was declared as required. Otherwise there shouldn't be en error, because the module is optional anyway.